docs/source/savant_101/30_dm.rst

```diff
@@ -200,8 +224,65 @@ Within ``output`` you may also select only necessary objects by specifying their
             kwargs:
               confidence_threshold: 0.1
 
-All skipped classes will be permanently excluded from the next steps of the pipeline. The ``selector`` block also allows defining a filter to eliminate unnecessary objects.
+All skipped classes will be permanently excluded from the next steps of the pipeline. 
+The ``selector`` block also allows defining a filter to eliminate unnecessary objects.
+
+If unit name is ``Primary_Detector``, then to address selected objects in the following units use ``Primary_Detector.person`` 
+and ``Primary_Detector.face`` labels.
+
+The default selector implementation runs NMS and allows selecting objects by specifying ``min_width``, ``min_height``, and ``confidence_threshold``. 
+To create a custom ``selector`` you have to implement :py:class:`~savant.base.selector.BaseSelector`. 
+You may want to take a look at :py:class:`~savant.selector.detector.BBoxSelector` to get an idea of how to write it.
+
+Converter vs Selector
+---------------------
+
+The converter and selector are two different concepts. The converter is used to convert the model output to a format that can be used by the selector. 
+The selector is used to select the objects that are of interest. Their functionality can duplicate each other, but they are used for different purposes.
+
+It mostly depends on the purpose, but as a rule of thumb, you should use the converter if you need to convert the model output to a format that can be used by the selector.
+You should use the selector if you need to filter out the objects that are of interest.
+
+.. note::
+
+    Despite their different roles, a custom implementation of the converter can contain features normally implemented in the selector. 
+    It depends on the developer's preferences and the performance considerations.
+
+
+Performance Considerations
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-If unit name is ``Primary_Detector``, then to address selected objects in the following units use ``Primary_Detector.person`` and ``Primary_Detector.face`` labels.
+The goal of the converter is to transform the model output to a format that can be used by the selector and exclude what is not required by running NMS 
+and exclude classes. This is important to avoid creating unnecessary metadata. However, when every class requires additional filtering, these operations 
```

Comment:
I would consider the selection in the converter as class-agnostic. This is how NMS works for YOLO in the converter; it saves the most reliable results using intersection, but not classes. This has its pros and cons.